src/java/org/apache/cassandra/db/virtual/ExceptionsTable.java

```diff
@@ -0,0 +1,301 @@
+        }
+
+        return result;
+    }
+
+    @Override
+    public DataSet data(DecoratedKey partitionKey)
+    {
+        SimpleDataSet result = new SimpleDataSet(metadata());
+
+        synchronized (buffer)
+        {
+            String exceptionClass = UTF8Type.instance.getSerializer().deserialize(partitionKey.getKey());
+            LinkedHashMap<String, ExceptionRow> partition = buffer.get(exceptionClass);
+
+            if (partition != null)
+            {
+                for (Map.Entry<String, ExceptionRow> row : partition.entrySet())
+                    populateRow(result, exceptionClass, row.getKey(), row.getValue());
+            }
+        }
+
+        return result;
+    }
+
+    private void populateRow(SimpleDataSet result, String exceptionClass, String exceptionLocation, ExceptionRow row)
+    {
+        result.row(exceptionClass, exceptionLocation)
+              .column(COUNT_COLUMN_NAME, row.count)
+              .column(LAST_MESSAGE_COLUMN_NAME, row.message)
+              .column(LAST_STACKTRACE_COLUMN_NAME, row.stackTrace)
+              .column(LAST_OCCURENCE_COLUMN_NAME, row.occurence);
+    }
+
+    @Override
+    public void truncate()
+    {
+        synchronized (buffer)
+        {
+            buffer.clear();
+        }
+    }
+
+    public static class ExceptionRow
+    {
+        public String exceptionClass;
+        public String exceptionLocation;
+        public int count;
+        public String message;
+        public List<String> stackTrace;
+        public Date occurence;
+
+        /**
+         * @param exceptionClass    exception class of uncaught exception
+         * @param exceptionLocation location where that exception was thrown
+         * @param message           message of given exception
+         * @param stackTrace        whole stacktrace of given exception
+         * @param occurenceTime     time when given exception ocurred, in milliseconds from epoch
+         */
+        public ExceptionRow(String exceptionClass,
+                            String exceptionLocation,
+                            int count,
+                            String message,
+                            List<String> stackTrace,
+                            long occurenceTime)
+        {
+            this.exceptionClass = exceptionClass;
+            this.exceptionLocation = exceptionLocation;
+            this.count = count;
+            this.stackTrace = stackTrace;
+            this.message = message;
+            this.occurence = new Date(occurenceTime);
+        }
+
+        public static List<String> extractStacktrace(StackTraceElement[] stackTraceArray)
+        {
+            List<String> result = new ArrayList<>();
+
+            for (StackTraceElement element : stackTraceArray)
+                result.add(element.toString());
+
+            return result;
+        }
+    }
+
+    @VisibleForTesting
+    static class BoundedMap extends LinkedHashMap<String, LinkedHashMap<String, ExceptionRow>>
```

Comment:
it looks like we do not rely on LinkedHashMap logic, should we just use HashMap (with afterNodeInsertion override) instead or just invoke the cleanup logic manually? It would be more straightforward IMHO..